doc/reading/merge-split-vs-split-merge.md

```diff
@@ -0,0 +1,30 @@
+# Merge then split vs split then merge
+
+If I have a tensor `T0[I1, I2]`, are the following two schedules equivalent?
+
+- **Schedule 1:** `[I1, I2] -- split -> [I1, I2/4, 4] -- merge -> [I1*(I2/4), 4]`.
+- **Schedule 2:** `[I1, I2] -- merge -> [I1*I2] -- split -> [(I1*I2)/4, 4]`.
+
+Where the divisions above are all ceildiv. The answer is no.
+Why? Let's see an example where `I1` has extent `2`, and `I2` has extent `5`.
+
+For schedule 1, after schedule, the extents of the leaf domain will be `[2*2, 4]`.
+So for this schedule, we will be iterating the tensor as:
+
+```python
```

Comment:
What exactly is the indexing in this example? I expected that indexing of the tensor would look like this:
```
T[0, 0], T[0, 1], T[0, 2], T[0, 3]
T[0, 4], T[1, 0], T[1, 1], T[1, 2]
T[1, 3], T[1, 4], T[2, 0], T[2, 1]
T[2, 2], T[2, 3], T[2, 4], T[3, 0]
```
where the real data is from T[0,0] to T[1,4] with `I1`'s extend being `2` and `I2`'s extend being `5`. Anything above will be ignored/handled by indexing.

Reply:
The tensor will be iterated like this:

```python
for i in range(4):
  for j in range(4):
    i1 = i / 2
    i2 = (i % 2) * 4 + j
    if i1 < 2 and i2 < 5:
      T[i1, i2]
```

so it is indeed as shown in the doc. Just out of bound accesses will be predicated away.

Reply:
I will need to process this example, and how things are processed exactly.

Nevertheless, thanks for the explanation!